cmd/kubectl-gadget/top/ebpf.go

```diff
@@ -0,0 +1,296 @@
+				fmt.Errorf("this gadget can not filter by pod name"))
+		}
+		if params.Containername != "" {
+			return commonutils.WrapInErrInvalidArg("--containername / -c",
+				fmt.Errorf("this gadget can not filter by container name"))
+		}
+		if params.LabelsRaw != "" || len(params.Labels) > 0 {
+			return commonutils.WrapInErrInvalidArg("--selector / -l",
+				fmt.Errorf("this gadget can not filter by selector"))
+		}
+
+		ebpfNodeStats = make(map[string][]types.Stats)
+
+		if len(args) == 1 {
+			outputInterval, err = strconv.Atoi(args[0])
+			if err != nil {
+				return commonutils.WrapInErrInvalidArg("<interval>",
+					fmt.Errorf("%q is not a valid value", args[0]))
+			}
+		} else {
+			outputInterval = types.IntervalDefault
+		}
+
+		config := &utils.TraceConfig{
+			GadgetName:       "ebpftop",
+			Operation:        "start",
+			TraceOutputMode:  "Stream",
+			TraceOutputState: "Started",
+			CommonFlags:      &params,
+			Parameters: map[string]string{
+				types.MaxRowsParam:  strconv.Itoa(maxRows),
+				types.IntervalParam: strconv.Itoa(outputInterval),
+				types.SortByParam:   sortBy,
+			},
+		}
+
+		// when params.Timeout == interval it means the user
+		// only wants to run for a given amount of time and print
+		// that result.
+		singleShot := params.Timeout == outputInterval
+
+		// start print loop if this is not a "single shoot" operation
+		if singleShot {
+			ebpfPrintHeader()
+		} else {
+			ebpfStartPrintLoop()
+		}
+
+		if err := utils.RunTraceStreamCallback(config, ebpfCallback); err != nil {
+			return commonutils.WrapInErrRunGadget(err)
+		}
+
+		if singleShot {
+			ebpfPrintEvents()
+		}
+
+		return nil
+	},
+	SilenceUsage: true,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		var err error
+		ebpfSortBy, err = types.ParseSortBy(sortBy)
+		if err != nil {
+			return commonutils.WrapInErrInvalidArg("--sort", err)
+		}
+
+		return nil
+	},
+	Args: cobra.MaximumNArgs(1),
+}
+
+func init() {
+	addTopCommand(ebpfCmd, types.MaxRowsDefault, types.SortBySlice)
+}
+
+func ebpfCallback(line string, node string) {
+	mutex.Lock()
+	defer mutex.Unlock()
+
+	var event types.Event
+
+	if err := json.Unmarshal([]byte(line), &event); err != nil {
+		fmt.Fprintf(os.Stderr, "Error: %s", commonutils.WrapInErrUnmarshalOutput(err, line))
+		return
+	}
+
+	if event.Error != "" {
+		fmt.Fprintf(os.Stderr, "Error: failed on node %q: %s", event.Node, event.Error)
+		return
+	}
+
+	ebpfNodeStats[node] = event.Stats
+}
+
+func ebpfStartPrintLoop() {
+	go func() {
+		ticker := time.NewTicker(time.Duration(outputInterval) * time.Second)
+		ebpfPrintHeader()
+		for {
+			_ = <-ticker.C
+			ebpfPrintHeader()
+			ebpfPrintEvents()
+		}
+	}()
+}
+
+func ebpfPrintHeader() {
+	switch params.OutputMode {
+	case commonutils.OutputModeColumns:
+		if term.IsTerminal(int(os.Stdout.Fd())) {
+			utils.ClearScreen()
+		} else {
+			fmt.Println("")
+		}
+		fmt.Printf("%-16s %-8s %-16s %-16s %-7s %-20s %-12s %-6s\n",
```

Comment:
BTW, you are not using the same format here for the time column: %s vs %v (when using custom-columns)